### PR TITLE
Escape element identifiers in picard

### PR DESCRIPTION
**Pull request by mvdbeek**
ping @jmchilton  @bgruening

### CODE REVIEW

**Comment by bgruening** [comment]
John is this really needed? Why can this not be handled by Galaxy.

File: tools/picard/picard_macros.xml [hunk 1]
```diff
@@ -17,7 +17,9 @@
     </xml>
 
     <token name="@symlink_element_identifier@"><![CDATA[
-    ln -f -s '$inputFile' '$inputFile.element_identifier' &&
+    #import re
+    #set escaped_element_identifier = re.sub('[^\w\-_]', '_', str($inputFile.element_identifier))
```
Comment (nsoranzo) [inline tools/picard/picard_macros.xml]: The underscore should already be included in `\w`, see https://docs.python.org/2/howto/regex.html#matching-characters

**Comment by jmchilton** [comment]
@bgruening If the framework does it - it might mess up analyses that are keyed on this correctly - it is like Excel guessing what datatype your input is and adjusting it. How it is escaped could depend on how it is used in the tool wrapper (quote types, configfile vs command, etc...). If you want to implement a ``$input.element_identifier_shell_escaped``` or something I'd be happy to do that.

**Comment by mvdbeek** [comment]
> @bgruening If the framework does it - it might mess up analyses that are keyed on this correctly - it is like Excel guessing what datatype your input is and adjusting it. How it is escaped could depend on how it is used in the tool wrapper (quote types, configfile vs command, etc...). If you want to implement a ``$input.element_identifier_shell_escaped``` or something I'd be happy to do that.

In that particular case the framework should just do the (escaped) symlinking.

**Comment by bgruening** [comment]
I think more framework support for these kind of stuff would be nice yes!

**Comment by bgruening** [comment]
Thanks Marius!

**Pull request merged**